src/db/service/ExecutionService.ts

```diff
@@ -28,6 +28,31 @@ export default class ExecutionService {
     this.repository = AppDataSource.getRepository(ExecutionEntity);
   }
 
+  private entityToModel(entity: ExecutionEntity): ExecutionModelType {
+    return instanceToPlain(entity) as unknown as ExecutionModelType;
+  }
+
+  public async fetch(): Promise<ExecutionModelType[]> {
+    const entities = await this.repository.find({
+      order: {
+        timestamp: 'DESC',
+      },
+    });
+    return entities.map((entity) => {
+      return this.entityToModel(entity);
+    });
+  }
+
+  public async getMostRecent(): Promise<ExecutionModelType> {
```

Comment:
This would actually be awesome if it had an optional parameter getMostRecent(n) where if "n" is defined it gets the n most recent executions, otherwise if not given it fetches all